lib/executor.js

```diff
@@ -171,10 +171,48 @@ function applyAppConfig(app, instructions) {
 
 function setupDataSources(app, instructions) {
   forEachKeyedObject(instructions.dataSources, function(key, obj) {
+    var opts = {
+      // do we need to limit dynamic-able keys? or everything can be dynamic
+      keys: ['host', 'url', 'port']
```

Comment:
Everything should be dynamic. I think this feature should not be tied to datasources only, it should work at least for  middleware and component configuration too.